src/paperoni/refinement/llm_process_affiliation/model.py

```diff
@@ -0,0 +1,35 @@
+from dataclasses import dataclass
+from pathlib import Path
+
+import gifnoc
+
+from ...model.classes import InstitutionCategory
+from ..llm_common import Explained, PromptConfig
+
+
+@dataclass
+class Affiliation:
+    # The normalized academic affiliation name
+    name: Explained[str]
+    # The institution category
+    category: Explained[InstitutionCategory]
+    # The country of the institution
+    country: Explained[str]
+
+
+@dataclass
+class Analysis:
+    # The normalized academic affiliation names with category and country
+    affiliations: list[Affiliation]
+
+
+DEFAULT_SYSTEM_MESSAGE = (Path(__file__).parent / "system-prompt.md").read_text()
+FIRST_MESSAGE = """### The academic affiliation names to process:
+{}"""
+
+
+llm_config: PromptConfig = gifnoc.define(
+    f"paperoni.{Path(__file__).parent.name}",
```

Comment:
I'm not sure that's a good idea, there is a risk of clash if it ends up at the same level as other such files.

Reply:
I was originally thinking about enclosing all of the structured models parts in a single file, switching the file would load a different structured model or a different version of the structured model. But yes if we extract the system-prompt for readability then there should not have multiple structured model in a single directory. Not sure how to enforce this tho.

```
structured_models/
+ llm_tool_1/
| +- model.py
| +- system-prompt.md
+ llm_tool_2/
| +- model.py
| +- system-prompt.md
```

Reply:
The latter, yes, and also if it was directly under `paperoni/refinement` the key would become `paperoni.refinement`. Basically you can only do this in one file per directory, and it's also easy to miss that if you rename the directory you also have to change the config.